### PR TITLE
fix(v2): rewrite clone

### PR DESCRIPTION
**Pull request by eranhirsch**
I'm procrastinating releasing v2 so in the meantime I took a look at clone.

I deprecated the support for `.clone()` function inside the cloned object. It wasn't part of the documentation and felt like it doesn't fit in the library.

### CODE REVIEW

File: src/clone.ts [hunk 1]
```diff
@@ -1,68 +1,110 @@
...
-  return value != null && typeof (value as any).clone === "function"
-    ? (value as any).clone()
-    : cloneImplementation(value, [], []);
+export function clone(): <T>(data: T) => T;
+
+export function clone(...args: ReadonlyArray<unknown>): unknown {
+  return purry(cloneImplementation, args);
 }
 
 // from https://github.com/ramda/ramda/blob/master/source/internal/_clone.js
```
Comment (cjquines) [inline src/clone.ts:39]: **nit:** we can remove this right?

File: src/clone.ts [hunk 2]
```diff
@@ -1,68 +1,110 @@
...
+  }
+
+  if (
+    typeof value !== "object" ||
+    value === null ||
+    value instanceof Date ||
+    value instanceof RegExp
+  ) {
+    // We can use the built-in deep cloning function.
+    return structuredClone(value);
```
Comment (cjquines) [inline src/clone.ts:57]: **thought:** for some reason i thought structuredClone had bad compat but it's fine

**Review by cjquines — APPROVED** [review]

File: src/clone.ts (hunk 2 above)
Comment (eranhirsch) [reply src/clone.ts:57]: Our tests run on Node 18 too, so if they pass i'm going to assume that the support is adequete for our use-case.

**Pull request merged**